modules/afsocket/afinet-dest.c

```diff
@@ -414,12 +470,25 @@ afinet_dd_queue(LogPipe *s, LogMessage *msg, const LogPathOptions *path_options,
   log_dest_driver_queue_method(s, msg, path_options, user_data);
 }
 
+static void
+_free_server_candidates(AFInetDestDriver *self)
+{
+  while (self->server_candidates)
+    {
+      g_free(self->server_candidates->data);
+      self->server_candidates = g_list_delete_link(self->server_candidates, self->server_candidates);
+    }
+}
+
 void
 afinet_dd_free(LogPipe *s)
 {
   AFInetDestDriver *self = (AFInetDestDriver *) s;
 
-  g_free(self->hostname);
+  if (self->server_candidates)
```

Comment:
with g_list_free full, this if is probably not needed. I ran through the latest greatest glib code, and as I see it handles null well.

Reply:
@furiel: thanks for comments
( keep in mind that I just ported what I found in another version and will fix only on next week the comments :) )